x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_management/components/action_delete/delete_button_modal.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  EuiConfirmModal,
+  EuiOverlayMask,
+  EuiSwitch,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EUI_MODAL_CONFIRM_BUTTON,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { DeleteAction } from './use_delete_action';
+
+type DeleteButtonModalProps = Pick<
+  DeleteAction,
+  | 'closeModal'
+  | 'deleteAndCloseModal'
+  | 'deleteTargetIndex'
+  | 'deleteIndexPattern'
+  | 'indexPatternExists'
+  | 'isModalVisible'
+  | 'item'
+  | 'toggleDeleteIndex'
+  | 'toggleDeleteIndexPattern'
+  | 'userCanDeleteIndex'
+>;
+export const DeleteButtonModal: FC<DeleteButtonModalProps> = ({
+  closeModal,
+  deleteAndCloseModal,
+  deleteTargetIndex,
+  deleteIndexPattern,
+  indexPatternExists,
+  isModalVisible,
+  item,
+  toggleDeleteIndex,
+  toggleDeleteIndexPattern,
+  userCanDeleteIndex,
+}) => {
+  if (item === undefined) {
+    return null;
+  }
+
+  const indexName = item.config.dest.index;
+
+  return (
+    <>
+      {isModalVisible && (
```

Comment:
I guess it would be cleaner if a parent component will be responsible for hiding the modal

Reply:
Updated in 4382fee.